### ISSUE TITLE
Remote Control via Desktop Sharing

### ISSUE THREAD

**Issue by bigbluebutton-issue-import** [bigbluebutton/bigbluebutton#1428]
Originally reported on Google Code with ID 673

```
Just as with the request for multi-user simultaneous sharing, what about having an option
for a presenter to give away control of desktop to another attendee (when making them
the presenter possibly?).  Would seem very logical to have an option to allow viewers
to "contribute" to work being done on the presenter's machine, such as working through
a problem.


```

Reported by `matthew%mpwita.com@gtempaccount.com` on 2010-10-21 19:09:30

**Comment by tobiasgoecke** [comment]
+1

**Comment by ffdixon** [comment]
We're shifting to a pure HTML5 client that is will use the browser's capabilities to send/receive audio and video.  With the HTML5 client, there will not be a need to install any plugin.  

However, the browsers do not give you the ability to remotely control the user's mouse and keyboard -- for good reason!

As such, for BigBlueButton 2.2-beta, which we're working on now, we won't have any ability to remotely control a computer.

We'll leave this open so we can investigate for future versions of BigBlueButton.

**Comment by Josecarpeniet** [comment]
Hi @ffdixon,
Vectera has a cool feature that would be nice at least for collaboration between users. They call it cobrowsing. You can see it here https://help.vectera.com/en/articles/3347286-how-do-you-set-up-co-browsing

It basically opens a shared browser screen directly on the application. That way, all attendees can use their mouse and keyboard here. 
It's not desktop control, but it would be fantastic to have it. Nowadays almost everything is on the cloud so this might be a feasible feature that would be more than enough.

**Comment by uka-support** [comment]
#8929 might cover this requirement partially: if there was an ability to point at things (or mark things), you're almost at remote control but without the security risk. Requires a second screen, though.

See my comment there: Real remote control via browser is pretty much impossible for security reasons.

**Comment by larsbrinkhoff** [comment]
I have a use case for this.  I'd like to exhibit retro games.  I would share the game window and then allow users to send keyboard events to play the game.

**Comment by frankneumann** [comment]
I have just tried this feature in Zoom and I would like to see it also in BBB. As this feature would be my only reason for using Zoom in exercise with students.
Any chances to get this implemented?

**Comment by matiasilva** [comment]
There was someone interested and who I believe even successfully implemented this in the developer mailing list, though I have not heard from them again.

**Comment by frankneumann** [comment]
Thanks, so you mean there might already be something in the github repo?
Could you please send me the link to the post in developer mailing list.

**Comment by matiasilva** [comment]
No, I don't think they ever implemented it. The thread to follow is this one: https://groups.google.com/g/bigbluebutton-dev/c/_Mtmbv6kuOY/m/Zt6P0OkbAgAJ. It would be nice to get an update and see how we can get their code into the code base.

**Comment by frankneumann** [comment]
I looked on the fork of the github repo of Brent W. Baccala: https://github.com/BrentBaccala/bigbluebutton 
He implements a sharing based on VNC. Let's see whether that works out...

**Comment by GhaziTriki** [comment]
I confirm that BrentBaccala/bigbluebutton is made with VNC and it works 😄

**Comment by Josecarpeniet** [comment]
@frankneumann and @GhaziTriki thanks for letting us know. 
I don't understand how we can use the fork of BrentBaccala/bigbluebutton and at the same time get the updates from the original version here  🤔 
Any guidance on how to proceed here guys, please?

**Comment by digitalrevisor** [comment]
Hi. We are sourching for remote support control software added to bigbluebutton. Thanks for sharing this feature. Has anyone tested a solution on bbb?

**Comment by thomas-mc-work** [comment]
Activity 👻

**Comment by ScriptFanix** [comment]
Hi. At HAProxy Technologies, we regularly have screen-sharing sessions with our customers, and sometimes have to take control of the customer's screen. For this, we currently rely on Zoom, which is a bit of a shame for a company publishing free software.
Having this feature in BBB would allow us to finally stop using Zoom. Without this, BBB cannot be an alternative to Zoom.

**Comment by ffdixon** [comment]
Remote control of the user's computer would need an extension -- likely some binary, platform specific code -- as the browsers do not have permissions to control to the user's computer.  The core team doesn't have any plans to implement this in the near future.

**Comment by thomas-mc-work** [comment]
How much effort would it cost to use the code of the Jitsi desktop app and adapt it to Mattermost? If possible at all.

**Comment by ffdixon** [comment]
We (the core developers) would welcome the answer if someone familiar with those components to explore adapting it to BigBlueButton.

**Comment by SAShapeShifter** [comment]
Any update of if there is a plugin or addition to roadmap for this feature
